advanced_alchemy/_listeners.py

```diff
@@ -0,0 +1,26 @@
+"""Application ORM configuration."""
+
+from __future__ import annotations
+
+from datetime import datetime, timezone
+from typing import Any
+
+from sqlalchemy.event import listens_for
+from sqlalchemy.orm import Session
+
+
+@listens_for(Session, "before_flush")
```

Comment:
Should this also listen for `AsyncSession`?

Reply:
I believe it should also listen for the async session, as it emits the same events, but I'm not entirely sure.